ts/tests/metrics_yaml_testing/metric_cache_unit_test.py

```diff
@@ -0,0 +1,215 @@
+"""
+Unit testing for MetricsCache class and yaml parsing.
+"""
+import unittest
+import sys
+import yaml
+from ts.metrics.metric import Metric
+from ts.metrics.dimension import Dimension
+from ts.metrics.metric_cache import MetricsCache
+
+
+class TestAddMetrics(unittest.TestCase):
```

Comment:
Before you spend too much time on tests, was wondering if we could instead use `pytest` which we leverage for most of our existing python tests